extensions/scarb-doc/src/docs_generation/markdown/traits.rs

```diff
@@ -264,22 +307,40 @@ pub fn mark_duplicated_item_with_relative_path<'a, T: TopLevelMarkdownDocItem +
         .collect::<Vec<_>>()
 }
 
-pub fn generate_markdown_list_for_top_level_subitems<T: TopLevelMarkdownDocItem>(
+pub fn generate_markdown_table_summary_for_top_level_subitems<T: TopLevelMarkdownDocItem>(
     subitems: &[&T],
-    header_level: usize,
+    context: &MarkdownGenerationContext,
+    link_to_index_summary: bool,
```

Comment:
the idea is to avoid linking to the all module page `modules.md` from any submodule because that would be misleading.  However, in the current proposition we do link only from the all-module-page back to the very same all-module-page. So maybe it doesn't make any sense to link that section at all ever??